app/models/requests/DataRequest.scala

```
@@ -56,6 +57,15 @@ final case class SubscriptionDataRequest[A](
   isAgent:               Boolean
 ) extends WrappedRequest[A](request)
 
+final case class ObligationsAndSubmissionsSuccessDataRequest[A](
+  request:                              Request[A],
+  userId:                               String,
+  subscriptionLocalData:                SubscriptionLocalData,
+  obligationsAndSubmissionsSuccessData: ObligationsAndSubmissionsSuccess,
+  enrolments:                           Set[Enrolment],
+  isAgent:                              Boolean = false
```

the default value can result in silent failures, we shouldn't face a scenario where we don't know whether the user is an Agent or not